libraries/AP_Landing/AP_Landing_Slope.cpp

```diff
@@ -66,6 +68,22 @@ bool AP_Landing::type_slope_verify_land(const Location &prev_WP_loc, Location &n
         }
     }
 
+    
+    // Check for weight-on-wheels touchdown
+    if (!touched_down) {
+        const bool touchdown_alt_override = !is_zero(touchdown_altitude) &&
+                (height < touchdown_altitude);
+        const bool touchdown_debounced = (AP_LandingGear::instance()->get_wow_state() == AP_LandingGear::LG_WOW) &&
+                AP_LandingGear::instance()->get_wow_state_duration_ms() >= 500;
+
+        if (touchdown_alt_override || touchdown_debounced) {
+            touched_down = true; // sticky flag
+            type_slope_stage = SLOPE_STAGE_FINAL; // force flare
```

Comment:
I don't like that you trigger the flare like this. It messes up the announcement stuff later on. This state should only be set in one place. So, let's make a flag to force the flare. Here's a commit I just wrote that you can cherry pick
https://github.com/magicrub/ardupilot/commit/c888b6ca7e309c39d0e6937cfa7ea5deb0c7c4dc

Reply:
Done